nomenclature/codelist.py

```diff
@@ -364,6 +335,35 @@ def read_excel(cls, name, source, sheet_name, col, attrs=None):
 
         return cls(name=name, mapping=mapp)
 
+    def check_illegal_characters(
+        self, config: NomenclatureConfig = None
+    ) -> Dict[str, Code]:
+        """Check that no illegal characters are left in codes after tag replacement"""
+        forbidden = ["{", "}"]
+        if config and config.illegal_characters:
+            forbidden += config.illegal_characters
+
+        def _check_string(value):
+            if isinstance(value, str):
+                if any(char in value for char in forbidden):
+                    raise ValueError(
+                        f"Unexpected character in {self.name}: '{code.name}'."
+                        " Check for illegal characters and/or if tags were spelled correctly."
+                    )
+            elif isinstance(value, dict):
+                for v in value.values():
+                    _check_string(v)
+            elif isinstance(value, list):
+                for item in value:
+                    _check_string(item)
+
+        for code in self.mapping.values():
+            if not code.repository:
+                for attr in code.model_fields:
+                    if attr not in ["file", "repository"]:
+                        value = getattr(code, attr)
+                        _check_string(value)
```

Comment:
This could be one line
```suggestion
                        _check_string(getattr(code, attr))
```

Reply:
Similar to my previous comment, you marked this as resolved but did not implement it or provide a reason why you did not implement it.

Reply:
Similar to my previous comment, you marked this as resolved but did not implement it or provide a reason why you did not implement it.